api4/group.go

```diff
@@ -575,23 +575,33 @@ func getGroups(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Err = model.NewAppError("Api4.getGroups", "api.ldap_groups.license_error", nil, "", http.StatusNotImplemented)
 		return
 	}
+	teamID := c.Params.NotAssociatedToTeam
+	channelID := c.Params.NotAssociatedToChannel
+
+	if len(teamID) <= 0 && len(channelID) <= 0 {
```

Comment:
hmm can we just instead do the following?

```
teamID := c.Params.NotAssociatedToTeam
channelID := c.Params.NotAssociatedToChannel

if !model.IsValidId(teamID) && !model.IsValidId(channelID) {
c.Err = model.NewAppError("Api4.getGroups", "api.getGroups.invalid_or_missing_channel_or_team_id", nil, "", http.StatusBadRequest)
		return
	}

```

This would simplify the validation logic above.

Reply:
That only tells you if both parameters are invalid, but you’d still have to invoke `IsValidId` again in each of the branching statements below to know whether to invoke `GetChannel` and `GetTeam`. With my proposal you can just test if they’re empty strings or not.

Reply:
Ahh fair enough. Makes sense. 👍 to your suggestions.